lua/tardis/cl_hud.lua

```diff
@@ -2,41 +2,73 @@
 
 surface.CreateFont("TARDIS-HUD-Large", {
     font="Roboto",
-    size=68
+    size=58
 })
 
 surface.CreateFont("TARDIS-HUD-Med", {
     font="Roboto",
-    size=40
+    size=36
 })
 
 surface.CreateFont("TARDIS-HUD-Small", {
     font="Roboto",
-    size=34
+    size=30
 })
 
-hook.Add("HUDPaint", "TARDIS-HUD", function()
-    if not TARDIS:GetSetting("health-enabled") then return end
-    if not (LocalPlayer():GetTardisData("interior") or LocalPlayer():GetTardisData("exterior")) then return end
-    local tardis = LocalPlayer():GetTardisData("exterior")
-    if not IsValid(tardis) then return end
-    local sw = ScrW()
-    local sh = ScrH()
-    local health = math.ceil(tardis:GetHealthPercent())
+local function CreatePercentageHUDPanel(text, value, offset, red_level)
+
+    red_level = red_level or 20
+
     local width = 115
-    if health >= 10 then width = width + 10 end
-    if health == 100 then width = width + 35 end
-    local height = (sw >= 800) and 120 or 95
-    local healthfont = (height == 120) and "TARDIS-HUD-Large" or "TARDIS-HUD-Med"
-    local x = (ScrW()-width)*0.02
-    local y = (ScrH()-height)*0.025
+    local height = (ScrW() >= 800) and 95 or 85
+
+    if value >= 10 then width = width + 10 end
+    if value == 100 then width = width + 20 end
+
+    local x = (ScrW() - width) * 0.02 + offset
+    local y = (ScrH() - height) * 0.025
+
+    local header_font = "TARDIS-HUD-Small"
```

Comment:
We can do some optimisations in this area like bringing out some of the static strings e.g. this one `header_font` as well as the others e.g. `TARDIS-HUD-Large/Med` which will reduce memory allocations, and calls to `NamedColor`

We can simply define them above as local variables outside the function scope and reference them within. I wouldn't normally bother with such micro optimisations but given this runs every frame potentially multiple times with artron and health enabled it's good to do optimise where possible

Reply:
befcee8d17bc313a91cad4e7940cbcbff17c1015